src/itertools.ts

```diff
@@ -201,9 +178,14 @@ export function* islice<T>(
     stop = stopOrStart;
   }
 
-  const pred = slicePredicate(start, stop, step);
+  if (start < 0) throw new Error("start cannot be negative");
+  if (stop !== null && stop < 0) throw new Error("stop cannot be negative");
+  if (step <= 0) throw new Error("step cannot be negative");
+
   for (const [i, value] of enumerate(iterable)) {
-    if (pred(i)) {
+    if (i < start) continue;
+    if (stop !== null && i >= stop) break;
+    if (step === 1 || (i - start) % step === 0) {
       yield value;
```

Comment:
can also inverse the condition to keep same code style as before with early continues/breaks
```js
if ((i - start) % step !== 0) continue;
yield value;
```

Reply:
I'm keeping it as-is, as to me the continue/break above select the start/end of the range, and this if-statement controls which values are skipped/emitted, which reads a bit clearer when using a positive condition.